.github/workflows/test.yml

```diff
@@ -39,55 +49,115 @@ jobs:
              run: ./housekeeping/makeBuild.sh -DCMAKE_TOOLCHAIN_FILE=cmake/san/clang-8_cxx17_ubsan.cmake
       name: "${{ matrix.options.name }}"
       runs-on: ubuntu-latest
-      container: soramitsu/kagome-dev:9 # Change to ${{ env.INDOCKER_IMAGE }} https://github.community/t/how-to-set-and-access-a-workflow-variable/17335/6
+      container: soramitsu/kagome-dev:10 # Change to ${{ env.INDOCKER_IMAGE }} https://github.community/t/how-to-set-and-access-a-workflow-variable/17335/6
       steps:
         - uses: actions/checkout@v2
+        - uses: actions/cache@v2
+          with:
+            path: ${{ env.CACHE_PATHS }}
+            key: ${{ github.job }}-${{ matrix.options.name }}-${{ env.CACHE_VERSION }}
         - name: "${{ matrix.options.name }}"
           run: "${{ matrix.options.run }}"
 
   clang-tidy:
       name: "Linux: clang-tidy"
       runs-on: ubuntu-latest
-      container: soramitsu/kagome-dev:9
+      container: soramitsu/kagome-dev:10
       steps:
         - uses: actions/checkout@v2
+        - uses: actions/cache@v2
+          with:
+            path: ${{ env.CACHE_PATHS }}
+            key: ${{ github.job }}-${{ env.CACHE_VERSION }}
         - name: clang-tidy
           env:
             # build only generated files, so clang-tidy will work correctl
             BUILD_FINAL_TARGET: generated
           run: |
             ./housekeeping/makeBuild.sh
             ./housekeeping/clang-tidy.sh
+
   coverage:
       name: "Linux: gcc-8 coverage/sonar"
       runs-on: ubuntu-latest
-      container: soramitsu/kagome-dev:9  # Change to ${{ env.INDOCKER_IMAGE }} https://github.community/t/how-to-set-and-access-a-workflow-variable/17335/6
+      container: soramitsu/kagome-dev:10  # Change to ${{ env.INDOCKER_IMAGE }} https://github.community/t/how-to-set-and-access-a-workflow-variable/17335/6
       steps:
         - uses: actions/checkout@v2
+        - uses: actions/cache@v2
+          with:
+            path: ${{ env.CACHE_PATHS }}
+            key: ${{ github.job }}-${{ env.CACHE_VERSION }}
         - name: makeBuild
           env:
             BUILD_FINAL_TARGET: ctest_coverage
           run: ./housekeeping/makeBuild.sh -DCMAKE_TOOLCHAIN_FILE=cmake/toolchain/gcc-8_cxx17.cmake -DCOVERAGE=ON
-        - name: Submit Coverage
+        - if: github.repository == 'soramitsu/kagome'
```

Comment:
you can't Submit Coverage from fork because ${{ secrets.CODECOV_TOKEN }} is  not available   https://docs.github.com/en/actions/configuring-and-managing-workflows/creating-and-storing-encrypted-secrets#using-encrypted-secrets-in-a-workflow